src/Symfony/Component/Workflow/Registry.php

```diff
@@ -23,19 +24,34 @@ class Registry
 
     /**
      * @param Workflow $workflow
-     * @param string   $className
+     * @param string|SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $className)
+    public function add(Workflow $workflow, $supportStrategy)
     {
-        $this->workflows[] = array($workflow, $className);
+        if (!$supportStrategy instanceof SupportStrategyInterface) {
```

Comment:
Thanks to @xabbuh I figured out the `@group legacy` annotation. Should be fine now :)